### PR TITLE
Connie sprint 2

### PR DESCRIPTION
**Pull request by conniejmiller**
This includes the previous fixes to the current master branch and my sprint 2 code.

### CODE REVIEW

File: validate.py [hunk 1]
```diff
@@ -91,55 +87,78 @@ def validate_dates(indi_list, fam_list):
         if not valid_month(birth_date):
             print('Error US42: Invalid birth month for ' +
                   row["NAME"] + ' (' + row["ID"] + ') ')
-        elif not date_compare(birth_date):
-            print('Error US01: Birth date of ' +
-                  row["NAME"] + ' (' + row["ID"] + ') ' +
-                  'occurs after the current date.')
+        else:
+            if not date_compare(birth_date,''):
+                print('Error US01: Birth date of ' +
```
Comment (ryancoppa) [inline validate.py:92]: Unrelated to this pull-request, but I was trying to think of a way to generalize our prints, they all have a common pattern... Just something to ponder.

**Review by ryancoppa — APPROVED** [review]
Looks solid

**Comment by vporta** [comment]
When I run the tests, this is my output: python test.py 
test_calculate_years (__main__.TestProject)
Testing the calculate_years function ... FAIL
test_date_compare (__main__.TestProject)
Testing the date_compare function ... ok
test_get_age (__main__.TestProject)
Testing the get_age function ... FAIL
test_get_birth (__main__.TestProject)
Testing the get_age function ... ok
test_get_name (__main__.TestProject)
Testing the get_age function ... ok
test_is_deceased (__main__.TestProject)
Testing the is_deceased function ... ok
test_life_duration (__main__.TestProject)
Testing if life duration is less than 150 years ... FAIL
test_valid_lifetime (__main__.TestProject)
Testing the valid_lifetime function ... ok
test_valid_month (__main__.TestProject)
Testing is month is valid ... ok
test_validate_genders (__main__.TestProject)
validate_genders(families, individuals): ... Anomaly US21: Wife Mary /Smith/ (@I2@) in family (@F1@) is not female.
ok

======================================================================
FAIL: test_calculate_years (__main__.TestProject)
Testing the calculate_years function
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 62, in test_calculate_years
    self.assertEqual(calculate_years('09 FEB 1962', '11 SEP 2017'), 55)
AssertionError: 56.0 != 55

======================================================================
FAIL: test_get_age (__main__.TestProject)
Testing the get_age function
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 42, in test_get_age
    self.assertEqual(get_age(individuals, '@I1@'), 117)
AssertionError: 118.0 != 117

======================================================================
FAIL: test_life_duration (__main__.TestProject)
Testing if life duration is less than 150 years
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test.py", line 35, in test_life_duration
    self.assertTrue(valid_lifetime('01 JAN 1000', '01 JAN 1149'))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 10 tests in 0.029s

FAILED (failures=3)

**Review by vporta — COMMENTED** [review]
I had 3 tests fail. Are you getting failing tests?

File: validate.py (hunk 1 above)
Comment (conniejmiller) [reply validate.py:92]: yes, I tackled that in my homework last week.  I'll try to do some refactoring.

**Pull request merged**